app/webpacker/components/CompetitionResultSubmission/UploadResultsJson.jsx

```diff
@@ -10,7 +10,12 @@ export default function UploadResultsJson({ competitionId, isWrtViewing }) {
   const [markResultSubmitted, setMarkResultSubmitted] = useCheckboxState(isWrtViewing);
 
   const { mutate: uploadResultsJsonMutate, error, isError } = useMutation({
-    mutationFn: () => uploadResultsJson({ competitionId, resultFile, markResultSubmitted }),
+    mutationFn: () => uploadResultsJson({
+      competitionId,
+      resultFile,
+      markResultSubmitted,
+      storeUploadedJson: !isWrtViewing, // The JSON will be uploaded to database only for Delegates.
```

Comment:
Will this lead to weird edge-cases when the regular, listed Delegate of a competition also happens to "accidentally" be a WRT member?

Reply:
No, the boolean is controlled from the page. If the page is 'submit results' of Delegates, then even for WRT member it will be `false`. It will be `true` only when the view is "admin page".

Reply:
Then maybe you should rename the variable to make that more clear. Something along the lines of `isWrtAdminView` (or perhaps even just `isAdminView`=

Reply:
Done